components/registry-facade/pkg/registry/blob.go

```diff
@@ -171,16 +171,27 @@ func (bh *blobHandler) getBlob(w http.ResponseWriter, r *http.Request) {
 		bp := bufPool.Get().(*[]byte)
 		defer bufPool.Put(bp)
 
-		n, err := io.CopyBuffer(w, rc, *bp)
+		const retryMax int = 5
+		var n int64
+		for i := 0; i <= retryMax; i++ {
+			n, err = io.CopyBuffer(w, rc, *bp)
+			if err == nil {
+				break
+			}
+		}
```

Comment:
I have some questions about the retry mechanism.
- Is there a reason that we retry any error? In other words, I wonder if we have to check what kind of error.
- Should we put the sleep before next challenge?

Reply:
In the last 7 days [log](https://cloudlogging.app.goo.gl/kHD1XC9YRsHkKwRf8), the errors are `connection reset by peer` and `broken pipe`. We should retry with these two errors only.